src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -172,15 +173,16 @@ public void UpdateTrackedDocument(Uri uri, SourceText newSourceText)
     {
         // Store the updated LSP view of the source text.
         Contract.ThrowIfFalse(_trackedDocuments.ContainsKey(uri), $"didChange received for {uri} which is not open.");
-        _trackedDocuments = _trackedDocuments.SetItem(uri, newSourceText);
+        var (_, language) = _trackedDocuments[uri];
+        _trackedDocuments = _trackedDocuments.SetItem(uri, (newSourceText, language));
 
         // If LSP changed, we need to compare against the workspace again to get the updated solution.
         _cachedLspSolutions.Clear();
 
         LspTextChanged?.Invoke(this, EventArgs.Empty);
     }
 
-    public ImmutableDictionary<Uri, SourceText> GetTrackedLspText() => _trackedDocuments;
+    public ImmutableDictionary<Uri, SourceText> GetTrackedLspText() => _trackedDocuments.ToImmutableDictionary(kvp => kvp.Key, kvp => kvp.Value.Text);
```

Comment:
I worry that if this is called on every request handling, we're now allocating new dictionaries each time which isn't great.

Reply:
Modified this to return the same dictionary (with the languageId).  That didn't seem unreasonable to me as the languageId is part of the tracked doc information.